nilearn/datasets/neurovault.py

```diff
@@ -0,0 +1,2589 @@
+        If more than `max_consecutive_fails` images or collections in
+        a row fail to be downloaded, we consider there is a problem
+        and stop the download session. Does not aplly to explicitely
+        specified images: the download will stop with an error message
+        if an image explicitly required by the caller fails to be
+        downloaded.
+
+    max_fails_in_collection : int, optional (default=5)
+        If more than `max_fails_in_collection` images fail to be
+        downloaded from a collection, we consider this collection is
+        bad and move on to the next collection.
+
+    batch_size : int, optional (default=None)
+        Neurovault sends metadata in batches. `batch_size` is the size
+        of the batches to ask for. If ``None``, the default
+        ``_DEFAULT_BATCH_SIZE`` will be used.
+
+    """
+    def __init__(self, neurovault_dir=None, download_mode='download_new',
+                 collection_terms=None, collection_filter=_empty_filter,
+                 image_terms=None, image_filter=_empty_filter,
+                 wanted_collection_ids=None, wanted_image_ids=None,
+                 download_manager=None, max_images=None,
+                 max_consecutive_fails=10, max_fails_in_collection=5,
+                 batch_size=None):
+        download_mode = download_mode.lower()
+        if download_mode not in ['overwrite', 'download_new', 'offline']:
+            raise ValueError(
+                'supported download modes are overwrite,'
+                ' download_new, offline; got {0}'.format(download_mode))
+        self.download_mode_ = download_mode
+        self.neurovault_dir_ = neurovault_directory(neurovault_dir)
+        if collection_terms is None:
+            collection_terms = {}
+        if image_terms is None:
+            image_terms = {}
+        if max_images is not None and max_images < 0:
+            max_images = None
+        self.collection_terms_ = collection_terms
+        self.collection_filter_ = collection_filter
+        self.image_terms_ = image_terms
+        self.image_filter_ = image_filter
+        self.visited_images_, self.visited_collections_ = set(), set()
+        self.max_images_ = max_images
+        self.max_consecutive_fails_ = max_consecutive_fails
+        self.max_fails_in_collection_ = max_fails_in_collection
+        self.consecutive_fails_ = 0
+        self.fails_in_collection_ = 0
+        self.batch_size_ = batch_size
+
+        if download_manager is None:
+            download_manager = BaseDownloadManager(
+                neurovault_data_dir=self.neurovault_dir_,
+                max_images=self.max_images_)
+        self.download_manager_ = download_manager
+        self.download_manager_.neurovault_data_dir_ = self.neurovault_dir_
+        self.wanted_image_ids_ = wanted_image_ids
+        self.wanted_collection_ids_ = wanted_collection_ids
+        self._prepare()
+
+    def _prepare(self):
```

Comment:
Minor style comment: this empty line makes things hard to read, because the separation of the new method is less visible.